service/internal/integrationtest/s3/presign_test.go

```diff
@@ -0,0 +1,156 @@
+// +build integration
+
+package s3
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/service/internal/integrationtest"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestInteg_PresignURL(t *testing.T) {
+	cases := map[string]struct {
+		body                 io.Reader
+		expires              time.Duration
+		sha256Header         string
+		expectedSignedHeader http.Header
+	}{
+		"standard": {
+			body: bytes.NewReader([]byte("Hello-world")),
+			expectedSignedHeader: http.Header{
+				"content-type":   {"application/octet-stream"},
+				"content-length": {"11"},
+			},
+		},
+		"nil-body": {
+			expectedSignedHeader: http.Header{},
+		},
+		"empty-body": {
+			body: bytes.NewReader([]byte("")),
+			expectedSignedHeader: http.Header{
+				"content-type": {"application/octet-stream"},
+			},
```

Comment:
With regard to the content-type, we should consider if empty body has the same behavior as nil body.